File: apps/agni-server/src/core/interactions/notification/toggleReadNotification.ts
import Repository from "@core/adapters/repository";
import { IUsecase } from "../interfaces";
import Notification from "@core/domains/entities/notification";

export class ToggleReadNotificationUseCase implements IUsecase<string, void> {
    private notificationRepo: Repository<Notification>

    constructor(notificationRepo: Repository<Notification>) {
        this.notificationRepo = notificationRepo
    }

    async execute(id: string): Promise<void> {
        const notification = await this.notificationRepo.get(id)
        if (notification) {
            notification.setIsRead(!notification.getIsRead())
            await this.notificationRepo.update(notification)
        }
    }
}

File: apps/agni_api/src/main/kotlin/dev/auguste/agni_api/core/usecases/notifications/ToggleReadNotification.kt
package dev.auguste.agni_api.core.usecases.notifications

import dev.auguste.agni_api.core.adapters.repositories.IRepository
import dev.auguste.agni_api.core.entities.Notification
import dev.auguste.agni_api.core.usecases.interfaces.IUseCase
import java.util.UUID

class ToggleReadNotification(val notificationRepo: IRepository<Notification>): IUseCase<UUID, Unit> {

    override fun execAsync(input: UUID) {
        val notification = notificationRepo.get(input)
        if (notification != null)  {
            notification.isRead = !notification.isRead
            notificationRepo.update(notification)
        }
    }
}